### PR TITLE
subsys: esb: pipefix and errata143fix

### PR DESCRIPTION
**Pull request by too1**

### CODE REVIEW

**Review by magnev — APPROVED** [review]
Looks good.
Tests are passing.

**Review by grochu — COMMENTED** [review]
Just added a minor editorial comment. @too1 should I also review from the protocol flow point of view? I am not very familiar with the ESB so far, so it would take some more time.

**Review by grochu — APPROVED** [review]
Looks good.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please fix style.

**Review by lemrey — COMMENTED** [review]
I am fine with the change.
Some lines are too long though, honestly I am surprised this is passing `checkpatch`

**Comment by grochu** [comment]
> I am fine with the change.
> Some lines are too long though, honestly I am surprised this is passing `checkpatch`

Agree, some lines could be split for better visibility. `Checkpatch` I think just generates warnings on the lengthy lines, so it won't fail and block the PR.

File: subsys/esb/esb.c [hunk 1]
```diff
@@ -168,6 +178,10 @@ static struct payload_rx_fifo rx_fifo;
 static uint8_t tx_payload_buffer[CONFIG_ESB_MAX_PAYLOAD_LENGTH + 2];
 static uint8_t rx_payload_buffer[CONFIG_ESB_MAX_PAYLOAD_LENGTH + 2];
 
+/* Random access buffer variables for ACK payload handling */
+struct payload_random_access_buf_wrapper ack_pl_container[CONFIG_ESB_TX_FIFO_SIZE];
+struct payload_random_access_buf_wrapper * ack_pl_container_entry_point_pr_pipe[CONFIG_ESB_PIPE_COUNT];
```
Comment (jfischer-no) [inline subsys/esb/esb.c]: You could rename this local `struct payload_random_access_buf_wrappe` to `pra_buf_wrapper` and  use `ack_pl[]` and `ack_pl_entry`. That would significantly mitigate the unreadability here.

**Comment by lemrey** [comment]
I understand the author is on parental leave so I am helping out.
I did some renaming to help readability.

File: subsys/esb/esb.c [hunk 2]
```diff
@@ -243,6 +243,48 @@ static uint32_t addr_conv(const uint8_t *addr)
 	return __REV(bytewise_bit_swap(addr));
 }
 
+static void apply_errata143_workaround(void)
```
Comment (nvlsianpu) [inline subsys/esb/esb.c:246]: static inline?

File: subsys/esb/esb.c [hunk 3]
```diff
@@ -243,6 +243,48 @@ static uint32_t addr_conv(const uint8_t *addr)
...
+{
+#ifdef CONFIG_SOC_NRF52832
```
Comment (nvlsianpu) [inline subsys/esb/esb.c:248]: If you would include `nrf_erratas.h` then you can leverage `NRF52_ERRATA_143_PRESENT` macro and function for runtime errata need test `nrf52_errata_143(void)`

**Comment by nvlsianpu** [comment]
@lemrey More SoC are affected by 143. Better to use MDK function and macro.

**Review by carlescufi — CHANGES_REQUESTED** [review]
@too1 please rebase to the latest master. I have a suspicion.

**Comment by lemrey** [comment]
Rebased

**Comment by lemrey** [comment]
Rebased again.
@nvlsianpu please have another look.

File: subsys/esb/esb.c (hunk 3 above)
Comment (lemrey) [reply subsys/esb/esb.c]: Ok

File: subsys/esb/esb.c (hunk 2 above)
Comment (lemrey) [reply subsys/esb/esb.c]: Ok

**Review by grochu — APPROVED** [review]

File: subsys/esb/esb.c [hunk 4]
```diff
@@ -229,6 +244,46 @@ static uint32_t addr_conv(const uint8_t *addr)
 	return __REV(bytewise_bit_swap(addr));
 }
```
Comment (nvlsianpu) [inline subsys/esb/esb.c:246]: #if NRF52_ERRATA_143_ENABLE_WORKAROUND

File: subsys/esb/esb.c [hunk 5]
```diff
@@ -229,6 +244,46 @@ static uint32_t addr_conv(const uint8_t *addr)
...
+			prefix0 == prefix7) {
+			/* This will cause a 3dBm sensitivity loss,
+			 * avoid using such address combinations if possible.
+			 */
+			*(volatile uint32_t *)0x40001774 =
+				((*(volatile uint32_t *)0x40001774) & 0xfffffffe) | 0x01000000;
+		}
+	}
+}
+
```
Comment (nvlsianpu) [inline subsys/esb/esb.c:286]: #endif NRF52_ERRATA_143_ENABLE_WORKAROUND

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
Compliance is fine now.

**Comment by ioannisg** [comment]
> Please fix style.

@jfischer-no @too1 is the above addressed?

**Review by jfischer-no — APPROVED** [review]

**Comment by jfischer-no** [comment]
> > Please fix style.
> 
> @jfischer-no @too1 is the above addressed?

small things are still there, but :see_no_evil:

**Pull request merged**